core/perf_test/PerfTest_Stream.cpp

```diff
@@ -0,0 +1,274 @@
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+/**
+ * @file PerfTest_Stream.cpp
+ * @brief Implementation of STREAM benchmark operations for Kokkos.
+ *
+ * @details This file provides a set of memory bandwidth benchmarks based on the
+ * STREAM benchmark suite. It implements the five core STREAM operations (Set,
+ * Copy, Scale, Add, and Triad) using Kokkos parallel primitives. It includes
+ * validation.
+ *
+ * The implementation strives to use as few Kokkos features as possible, and so
+ * validation is performed on the host rather than via parallel_reduce
+ */
+
+#include <Kokkos_Core.hpp>
+#include <benchmark/benchmark.h>
+#include "Benchmark_Context.hpp"
+
+constexpr static double aInit  = 1.0;
+constexpr static double bInit  = 2.0;
+constexpr static double cInit  = 3.0;
+constexpr static double scalar = 4.0;
+
+using StreamDeviceArray =
+    Kokkos::View<double*, Kokkos::MemoryTraits<Kokkos::Restrict>>;
+using StreamHostArray = typename StreamDeviceArray::HostMirror;
+
+// different than benchmarks/stream, which uses int
+// wide index types are common as GPU memory grows
+using StreamIndex = int64_t;
+using Policy      = Kokkos::RangePolicy<Kokkos::IndexType<StreamIndex>>;
+
+template <typename V>
+void perform_set(V& a, const double scalar_) {
+  Kokkos::parallel_for(
+      "set", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { a[i] = scalar_; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_copy(V& a, V& b) {
+  Kokkos::parallel_for(
+      "copy", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { b[i] = a[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_scale(V& b, V& c, const double scalar_) {
+  Kokkos::parallel_for(
+      "scale", Policy(0, b.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { b[i] = scalar_ * c[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_add(V& a, V& b, V& c) {
+  Kokkos::parallel_for(
+      "add", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { c[i] = a[i] + b[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_triad(V& a, V& b, V& c, const double scalar_) {
+  Kokkos::parallel_for(
+      "triad", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { a[i] = b[i] + scalar_ * c[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+int validate_array(V& a_dev, const double expected) {
+  auto a = Kokkos::create_mirror_view(a_dev);
```

Comment:
```suggestion
  const auto a = Kokkos::create_mirror_view(a_dev);
```

Reply:
(probably out of scope, but it may be worth finding out why `clang-tidy` is not detecting these issues of missing `const`)